### PR TITLE
[ADD] Background tag filtering UI

### PR DESCRIPTION
**Pull request by zorgjbe**
This makes the tag filter UI work using the new tags. It works, though it only filters one tag at the moment and the UI is pretty jank, so it's definitely a work-in-progress.

### CODE REVIEW

**Comment by Sekkmer** [comment]
image

File: pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.tsx [hunk 1]
```diff
@@ -507,18 +534,18 @@ function BackgroundSelectDialog({ hide, current, select }: {
 
 	const availableBackgrounds = useMemo(() => assetManager.getBackgrounds(), [assetManager]);
 	const [nameFilter, setNameFilter] = useState('');
-	/*
-	 * TODO: Add a tag based filter to the dialog in a later version
-	 * const [tagFilter, setTagFilter] = useState('');
-	 */
+	const [tags, setTags] = useState<readonly string[]>([...assetManager.backgroundTags.keys()]);
+	const knownTagCategories = useMemo(() => uniq([...assetManager.backgroundTags.values()].map((tag) => tag.category)), [assetManager.backgroundTags]);
```
Comment (Jomshir98) [inline pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.tsx:538]: Selection behavior:
Having all selected by default and unselecting feels very weird - one has to first unselect before selecting those that are wanted.
We think much better would be to have by default all unselected and have a special case that none selected is the same as all selected (empty filter is ignored)

Comment (Jomshir98) [inline pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.tsx:538]: Combining logic:
Having the tags combine with "OR" logic across categories is weird, especially with the fact that everything is selected by default - this basically means that one has to first deselect everything in all categories and then select the one thing one is searching for.
I think the categories should combine with an "AND" - categories where nothing is selected (which would be default) are ignored, items within category are combined with "OR" and categories themselves are combined with "AND".
Example selection:
- Time: Night
- Location: Bar, Beach
- Ambience, Item, Space: [nothing]

This selects all backgrounds that have "Night" tag _and_ have at least one of ["Bar", "Beach"] tags.

File: pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss [hunk 2]
```diff
@@ -63,11 +63,36 @@
 	gap: 0.5em;
```
Comment (Jomshir98) [inline pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss:63]: I think the `.backgroundSelect` needs its `max-width` and `max-height` changed to be constant, not max (`width`&`height`), as it feels really clunky when everything moves under your hand when modifying filters.
We think the following values look good on both normal screens and small ones (phones):
```css
width: min(50em, 90vw);
height: min(40em, 90vh);
```

File: pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss [hunk 3]
```diff
@@ -63,11 +63,36 @@
...
+			column-gap: 0.6em;
+			width: 100%;
+
+			span {
+				margin: auto 0;
+			}
+		}
+
+		.header-tags {
+			@include flex(row);
```
Comment (Jomshir98) [inline pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss:83]: This needs to wrap for narrow screens.
```suggestion
			@include flex(row, wrap);
```

File: pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss [hunk 4]
```diff
@@ -101,6 +128,7 @@
 		}
 
 		.dropdown-content {
+			@include flex(column);
 			display: none;
 			position: absolute;
```
Comment (Jomshir98) [inline pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss:133]: This element can easily overflow the screen for long lists. The suggested solution isn't perfect, but solves most of the overflowing:
```suggestion
			position: absolute;
			max-height: 50vh;
			overflow-y: auto;
```

**Review by Jomshir98 — CHANGES_REQUESTED** [review]

File: pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss (hunk 2 above)
Comment (Sekkmer) [reply pandora-client-web/src/components/chatroomAdmin/chatroomAdmin.scss:62]: I didn't quite like the result for the height so I moved the dialog to the top so the selection doesn't jump around

**Review by Jomshir98 — APPROVED** [review]
Looks good to me and feels great to use!

**Pull request merged**